docs/architecture/adr-053-state-sync-prototype.md

```diff
@@ -0,0 +1,196 @@
+# ADR 053: State Sync Prototype
+
+This ADR describes an initial state sync prototype, and is subject to change as we gain feedback and experience.
+
+## Changelog
+
+* 2020-01-28: Initial draft (Erik Grinaker)
+
+## Context
+
+State sync will allow a new node to receive a snapshot of the application state without downloading blocks or going through consensus. This bootstraps the node significantly faster than the current fast sync system, which replays all historical blocks.
+
+Background discussions and justifications are detailed in [ADR-042](./adr-042-state-sync.md), which can be summarized as:
+
+* The application periodically takes full state snapshots (i.e. eager snapshots).
+
+* The application splits snapshots into smaller chunks that can be individually verified against a chain Merkle root.
+
+* Tendermint uses the light client to obtain a trusted chain Merkle root for verification.
+
+* Tendermint discovers and downloads snapshot chunks in parallel from multiple peers, and passes them to the application via ABCI to be applied and verified against the chain root.
+
+* Historical blocks are not backfilled, so state synced nodes will have a truncated block history.
+
+## Tendermint Proposal
+
+This describes the snapshot/restore process seen from Tendermint. The interface is kept as small and general as possible to give applications maximum flexibility.
+
+### Snapshot Data Structure
+
+```proto
+message Snapshot {
+    uint64 height = 1;   // The height at which the snapshot was taken
+    uint64 format = 2;   // The application-specific snapshot format version
+    uint64 chunks = 3;   // The number of chunks in the snapshot
+    bytes metadata = 4;  // Arbitrary application metadata
+}
+
+message SnapshotChunk {
+    uint64 height = 1;   // The height of the corresponding snapshot
+    uint64 format = 2;   // The snapshot format version
+    uint64 chunk = 3;    // The chunk index (zero-based)
+    bytes data = 4;      // Serialized application state in an arbitrary format
+}
+```
+
+A node can have multiple snapshots taken at various heights, in different application-specific format versions. Each snapshot consists of multiple chunks containing the actual state data.
```

Comment:
Ohh I see. Maybe move this paragraph between the section subtitle and the proto definitions

Reply:
Done